functions/utils/getAsset.ts

```diff
@@ -0,0 +1,28 @@
+import { AssetDocument } from '../../src/graphql/data/__generated__/types-and-hooks'
+import client from '../client'
+
+export default async function getAsset(collectionAddress: string, tokenId: string, url: string) {
+  const { data } = await client.query({
+    query: AssetDocument,
+    variables: {
+      address: collectionAddress,
+      filter: {
+        tokenIds: [tokenId],
+      },
+    },
+  })
+  const asset = data?.nftAssets?.edges[0]?.node
+  if (!asset) {
+    return undefined
+  }
+  const formattedAsset = {
+    title: asset.name
+      ? asset.name
+      : asset.collection?.name
+      ? asset.collection.name + ' #' + asset.tokenId
+      : 'Asset #' + asset.tokenId,
```

Comment:
this is a little hard to read. maybe use the "nullish coalescing" operator (i.e. `??`) to cut down on visual branching, or just make a helper function

Reply:
also, maybe we should have a fallback title here if none of the information is found